### PR TITLE
add automl classification model card

### PR DESCRIPTION
**Pull request by sharma-riti**
Pipeline - https://ml.azure.com/experiments/id/6484ce93-6993-479d-8a7e-4f4950fcbc9e/runs/orange_jewel_1mr6xldk54?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Model card - https://ml.azure.com/registries/automl-images-dev/models/AutoML-Image-Classification/version/5?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

### CODE REVIEW

File: assets/models/system/automl-image-classification/description.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Image Multi-label classification|[multilabel fridgeObjects](https://cvbp-secondary.z19.web.core.windows.net/datasets/image_classification/multilabelFridgeObjects.zip)|<a href="https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/automl-standalone-jobs/automl-image-classification-multilabel-task-fridge-items/automl-image-classification-multilabel-task-fridge-items.ipynb" target="_blank">image-classification-multilabel.ipynb</a>|<a href="https://github.com/Azure/azureml-examples/blob/main/cli/jobs/automl-standalone-jobs/cli-automl-image-classification-multilabel-task-fridge-items/cli-automl-image-classification-multilabel-task-fridge-items.yml" target="_blank">image-classification-multilabel.yml</a>
+
+
+### Inference samples
```
Comment (raduk) [inline assets/models/system/automl-image-classification/description.md:34]: Should we point to the automl inference notebooks?

File: assets/models/system/automl-image-classification/description.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    {
+        "probs": [0.04, 0.93],
+        "labels": ["label1", "label2"]
+    }
+]
+```
+
+#### Model inference - visualization
+For a sample image below, the top 4 labels are 'trilobite', 'affenpinscher, monkey pinscher, monkey dog', 'mosque', 'Bernese mountain dog'.
```
Comment (raduk) [inline assets/models/system/automl-image-classification/description.md:78]: the highest one doesn't seem to match with the image ...

File: assets/models/system/automl-image-classification/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        Standard_D64as_v4, Standard_D96a_v4, Standard_D96as_v4, Standard_FX4mds, Standard_F8s_v2,
+        Standard_FX12mds, Standard_F16s_v2, Standard_F32s_v2, Standard_F48s_v2, Standard_F64s_v2,
+        Standard_F72s_v2, Standard_FX24mds, Standard_FX36mds, Standard_FX48mds, Standard_E2s_v3,
+        Standard_E4s_v3, Standard_E8s_v3, Standard_E16s_v3, Standard_E32s_v3, Standard_E48s_v3,
+        Standard_E64s_v3, Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3,
+        Standard_NC12s_v3, Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3,
+        Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4,
+        Standard_ND96asr_v4, Standard_ND96amsr_A100_v4, Standard_ND40rs_v2
+tags:
+   Preview: ''
```
Comment (bhimar) [inline assets/models/system/automl-image-classification/spec.yaml:21]: do we need preview if AutoML is GA?

File: assets/models/system/automl-image-classification/description.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+Automated machine learning, or AutoML, is a process that automates the repetitive and time-consuming tasks involved in developing machine learning models. This helps data scientists, analysts, and developers to create models more efficiently and with higher quality, resulting in increased productivity and scalability.
+AutoML Image Classification enables you to train machine learning models to perform image multiclass and multilabel tasks according to your own defined labels. It is a task of computer vision that involves analyzing and categorizing an image into specific label(s).
```
Comment (bhimar) [inline assets/models/system/automl-image-classification/description.md:2]: "It is a compute vision task"

File: assets/models/system/automl-image-classification/description.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+Data inputs can then be created using the Training and optional Validation MLTables.
+
+#### Train a Model
+
+One can initiate [individual trials](https://learn.microsoft.com/azure/machine-learning/how-to-auto-train-image-models?view=azureml-api-2&tabs=python#individual-trials), [manual sweeps](https://learn.microsoft.com/azure/machine-learning/how-to-auto-train-image-models?view=azureml-api-2&tabs=python#manually-sweeping-model-hyperparameters), or [automatic sweeps](https://learn.microsoft.com/azure/machine-learning/how-to-auto-train-image-models?view=azureml-api-2&tabs=python#automatically-sweeping-model-hyperparameters-automode). It is suggested to begin with an automatic sweep to establish a baseline model. Afterward, one can experiment with individual trials using specific models and hyperparameter configurations. Lastly, manual sweeps can be used to explore multiple hyperparameter values near the more promising models and hyperparameter configurations. This three-step process (automatic sweep, individual trials, manual sweeps) helps avoid searching the entirety of the hyperparameter space, which grows exponentially with the number of hyperparameters.
```
Comment (bhimar) [inline assets/models/system/automl-image-classification/description.md:22]: Replace "One" with "You" because we used "you" previously in the description

File: assets/models/system/automl-image-classification/description.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      "index":[0, 1],
+      "data":[
+         ["image1"],
+         ["image2"]
+      ]
+   }
+}
+```
+Note:
+- "image1" and "image2" should be strings in `base64` format.
```
Comment (bhimar) [inline assets/models/system/automl-image-classification/description.md:60]: do we support image url for automl?

File: assets/models/system/automl-image-classification/description.md (hunk 2 above)
Comment (bhimar) [reply assets/models/system/automl-image-classification/description.md:78]: Actually I'm not sure that any of these labels actually match the image (even the dog specie).

Comment (sharma-riti) [reply assets/models/system/automl-image-classification/description.md:78]: yep, that is why I wanted to use the fine-tuned images (as discussed yesterday). Results are not very good on the images from https://brandcentral.microsoft.com/. Please let me know if we are allowed to use any other images. I am only aware of brandcentral.

File: assets/models/system/automl-image-classification/description.md (hunk 6 above)
Comment (sharma-riti) [reply assets/models/system/automl-image-classification/description.md:60]: nope

File: assets/models/system/automl-image-classification/description.md (hunk 1 above)
Comment (sharma-riti) [reply assets/models/system/automl-image-classification/description.md:34]: Those notebooks are for local inferencing... I thought we should be providing deployments/inferencing using computes?

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/automl/conda.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+channels:
+- conda-forge
+dependencies:
+- python=3.8.17
+- pip<=23.2.1
+- pip:
+  - mlflow==2.6.0
+  - torch==1.13.1
+  - transformers==4.33.1
```
Comment (PhaniShekhar) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/automl/conda.yaml:9]: Maybe transformers and accelerate can be removed ?

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py [hunk 8]
```diff
@@ -746,3 +747,107 @@ def _prepare_artifacts_dict(self) -> Dict:
...
+        :rtype: mlflow.models.signature.ModelSignature
+        """
+
+        from azureml.automl.dnn.vision.common.model_export_utils import _get_mlflow_signature
+        return _get_mlflow_signature(self._task)
+
+
+    def save_as_mlflow(self):
+        """Prepare model for save to MLflow."""
+        sys.path.append(self.MODEL_DIR)
```
Comment (PhaniShekhar) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:829]: This is probably not needed since we are not importing anything from self.MODEL_DIR path

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: assets/models/system/automl-image-classification/description.md (hunk 1 above)
Comment (yiyoulin) [reply assets/models/system/automl-image-classification/description.md:34]: I see that the notebook is from foundation model, and the same story happens for od/ seg. We don't have a inference notebook for automl. I thought of dropping this part as the inference is in the finetune notebook last section, so we could do "sample notebook" instead of specifying for "finetune", "inference", what do you think?

File: assets/models/system/automl-image-classification/spec.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        Standard_D64as_v4, Standard_D96a_v4, Standard_D96as_v4, Standard_FX4mds, Standard_F8s_v2,
+        Standard_FX12mds, Standard_F16s_v2, Standard_F32s_v2, Standard_F48s_v2, Standard_F64s_v2,
+        Standard_F72s_v2, Standard_FX24mds, Standard_FX36mds, Standard_FX48mds, Standard_E2s_v3,
+        Standard_E4s_v3, Standard_E8s_v3, Standard_E16s_v3, Standard_E32s_v3, Standard_E48s_v3,
+        Standard_E64s_v3, Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3,
+        Standard_NC12s_v3, Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3,
+        Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4,
+        Standard_ND96asr_v4, Standard_ND96amsr_A100_v4, Standard_ND40rs_v2
+tags:
+   license: apache2.0
```
[the remark this answers is not in the document]
Comment (sharma-riti) [reply assets/models/system/automl-image-classification/spec.yaml:21]: apache2.0

**Review by PhaniShekhar — APPROVED** [review]

File: assets/models/system/automl-image-classification/description.md (hunk 1 above)
Comment (sharma-riti) [reply assets/models/system/automl-image-classification/description.md]: changed to "code samples"

**Review by yiyoulin — APPROVED** [review]

**Review by PhaniShekhar — APPROVED** [review]
